src/test/system/lender/ratio/two-tranches.sol

```diff
@@ -0,0 +1,150 @@
+        createTestUsers(deploySeniorTranche);
+
+        hevm = Hevm(0x7109709ECfa91a80626fF3989D68f67F5b1DD12D);
+        hevm.warp(1234567);
+    }
+
+    function testSupplyJuniorFirst() public {
+        uint seniorInvestorAmount = 400 ether;
+        uint juniorInvestorAmount = 100 ether;
+
+        // minJuniorRatio 20%
+        uint minJuniorRatio = 2 * 10**26;
+        AssessorLike(assessor).file("minJuniorRatio" , minJuniorRatio);
+
+        supplyJunior(juniorInvestorAmount);
+        supplySenior(seniorInvestorAmount);
+
+        // currency equals token amount
+        seniorInvestor.doRedeem(seniorInvestorAmount);
+        juniorInvestor.doRedeem(juniorInvestorAmount);
+    }
+
+    function testFailNotSupplyJuniorFirst() public {
+        uint seniorInvestorAmount = 400 ether;
+
+        // minJuniorRatio 20%
+        uint minJuniorRatio = 2 * 10**26;
+        AssessorLike(assessor).file("minJuniorRatio" , minJuniorRatio);
+
+        supplySenior(seniorInvestorAmount);
+    }
+
+    function testFailSupplyNotApproved() public {
+        uint seniorInvestorAmount = 401 ether;
+        uint juniorInvestorAmount = 100 ether;
+
+        // minJuniorRatio 20%
+        uint minJuniorRatio = 2 * 10**26;
+        AssessorLike(assessor).file("minJuniorRatio" , minJuniorRatio);
+
+        supplyJunior(juniorInvestorAmount);
+        // senior supply not approved
+        supplySenior(seniorInvestorAmount);
+    }
+
+    function testAdditionalSupply() public {
+        uint seniorInvestorAmount = 400 ether;
+        uint juniorInvestorAmount = 100 ether;
+
+        // minJuniorRatio 20%
+        uint minJuniorRatio = 2 * 10**26;
+        AssessorLike(assessor).file("minJuniorRatio" , minJuniorRatio);
+
+        supplyJunior(juniorInvestorAmount);
+        supplySenior(seniorInvestorAmount);
+
+        // additional junior supply
+        supplyJunior(50 ether);
+
+        // junior 150 ether max senior: 400 ether
+        supplySenior(200 ether);
+
+        assertEq(AssessorLike(assessor).currentJuniorRatio(), minJuniorRatio);
+    }
+
+    function testJuniorRedeemFirst() public {
+        uint seniorInvestorAmount = 400 ether;
+        uint juniorInvestorAmount = 200 ether;
+
+        // minJuniorRatio 20%
+        uint minJuniorRatio = 2 * 10**26;
+        AssessorLike(assessor).file("minJuniorRatio" , minJuniorRatio);
+
+        supplyJunior(juniorInvestorAmount);
+        supplySenior(seniorInvestorAmount);
+
+        juniorInvestor.doRedeem(100 ether);
+
+        assertEq(AssessorLike(assessor).currentJuniorRatio(), minJuniorRatio);
+    }
+
+    function testBreakRatioWithDefault() public {
+        uint seniorInvestorAmount = 400 ether;
+        uint juniorInvestorAmount = 100 ether;
+
+        // minJuniorRatio 20%
+        uint minJuniorRatio = 2 * 10**26;
+        AssessorLike(assessor).file("minJuniorRatio" , minJuniorRatio);
+
+        supplyJunior(juniorInvestorAmount);
+        supplySenior(seniorInvestorAmount);
+
```

Comment:
really good test.  I think it would make sense to pick a lower number for senior investment then 400 though (A value that is not max allowed senior investment). Because otherwise the additional supplyToSenior is already not permitted even before the default 🙈

Reply:
I wanted to test the edge cases :)